src/main/java/dev/cdevents/CDEventsGenerator.java

```diff
@@ -0,0 +1,129 @@
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+            mustache.execute(bufferedWriter, schemaData).flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new CDEventsException("Exception occurred while generating class file from Json schema ", e);
+        }
+        System.out.println("Rendered event-template has been written to file - "+classFile.getAbsolutePath());
+    }
+
+    private static SchemaData buildCDEventDataFromJsonSchema(File file) {
+        SchemaData schemaData = new SchemaData();
+
+        System.out.println("Processing event JsonSchema file: " + file.getAbsolutePath());
+        try {
+            JsonNode rootNode = objectMapper.readTree(file);
+            JsonNode contextNode = rootNode.get("properties").get("context").get("properties");
+
+            String eventType = contextNode.get("type").get("enum").get(0).asText();
+            System.out.println("eventType: " + eventType);
+            String[] type = eventType.split("\\.");
+            String subject = type[2];
+            String predicate = type[3];
+            String capitalizedSubject = StringUtils.capitalize(subject);
+            if(subject.equals("pipelinerun")){
+                capitalizedSubject = subject.substring(0, 1).toUpperCase() + subject.substring(1, 8) + subject.substring(8, 9).toUpperCase() + subject.substring(9);
+            }
+            String capitalizedPredicate = StringUtils.capitalize(predicate);
+            String version = type[4];
+
+            //set the Schema JsonNode required values to schemaData
+            schemaData.setSubject(subject);
+            schemaData.setPredicate(predicate);
+            schemaData.setCapitalizedSubject(capitalizedSubject);
+            schemaData.setCapitalizedPredicate(capitalizedPredicate);
+            schemaData.setSchemaFileName(file.getName());
+            schemaData.setUpperCaseSubject(subject.toUpperCase());
+            schemaData.setVersion(version);
+
+            JsonNode subjectNode = rootNode.get("properties").get("subject").get("properties");
+            JsonNode subjectContentNode = subjectNode.get("content").get("properties");
+            updateSubjectContentProperties(schemaData, subjectContentNode);
+        } catch (IOException e) {
+            throw new CDEventsException("Exception occurred while building schema data from Json schema ", e);
+        }
+        return schemaData;
+    }
+
+    private static void updateSubjectContentProperties(SchemaData schemaData, JsonNode subjectContentNode) {
+        Iterator<Map.Entry<String, JsonNode>> contentProps = subjectContentNode.fields();
+        List<SchemaData.ContentField> contentFields = new ArrayList<>();
+        List<SchemaData.ContentObjectField> contentObjectFields = new ArrayList<>();
+        for (Iterator<Map.Entry<String, JsonNode>> entryProps = contentProps; entryProps.hasNext(); ) {
```

Comment:
```suggestion
	    while (contentProps.hasNext()) {
            Map.Entry<String, JsonNode> contentMap = contentProps.next();
```
Could this for loop be replaced by a while loop for readability?

Reply:
agree, looks fine